### PR TITLE
Migrate banner_messages.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/banner_messages.dart [hunk 1]
```diff
@@ -25,13 +26,13 @@ const preCompileShadersDocsUrl =
 
 class BannerMessagesController {
   final _messages = <String, ValueNotifier<List<BannerMessage>>>{};
-  final _dismissedMessageKeys = <Key>{};
+  final Set<Key?> _dismissedMessageKeys = <Key?>{};
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/banner_messages.dart:29]: why do we need to add the type on the LHS? `<Key?>` should be enough

File: packages/devtools_app/lib/src/shared/banner_messages.dart [hunk 2]
```diff
@@ -423,8 +422,8 @@ void maybePushDebugModePerformanceMessage(
 ) {
   if (offlineController.offlineMode.value) return;
   if (serviceManager.connectedApp?.isDebugFlutterAppNow ?? false) {
-    Provider.of<BannerMessagesController>(context)
-        .addMessage(DebugModePerformanceMessage(screenId).build(context));
+    Provider.of<BannerMessagesController>(context).addMessage(
+        DebugModePerformanceMessage(screenId).build(context) as BannerMessage);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/banner_messages.dart:426]: nit: add trailing comma

File: packages/devtools_app/lib/src/shared/banner_messages.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/banner_messages.dart:29]: done

File: packages/devtools_app/lib/src/shared/banner_messages.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/banner_messages.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**